### ISSUE TITLE
no-spaced-func when above a 'Assignment without declaration' destructuring

### ISSUE THREAD

**Issue by chrisnojima** [babel/babel-eslint#187]
```
let a = 1
({a} = {a: 2})
console.log(a)
```

line 2 outputs: 
```
 1:9  error  Unexpected space between function name and paren  no-spaced-func

✖ 1 problem (1 error, 0 warnings)
```

Changing line 1 to
```
let a = 1;
```
stops this warning.

This should be correct syntax (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Assignment_without_declaration_2)

**Comment by sebmck** [comment]
It's not, the lint error is correct. Automatic semicolon insertion does not consider that to be a valid place for a semicolon so what you're basically writing is:

```javascript
let a = 1
({a} = {a: 2})
console.log(a)
```
